src/sentry/uptime/endpoints/validators.py

```diff
@@ -34,6 +48,19 @@ class UptimeMonitorValidator(CamelSnakeSerializer):
     )
     mode = serializers.IntegerField(required=False)
 
+    def validate_url(self, url):
+        url_parts = extract_domain_parts(url)
+        existing_count = ProjectUptimeSubscription.objects.filter(
+            uptime_subscription__url_domain=url_parts.domain,
+            uptime_subscription__url_domain_suffix=url_parts.suffix,
+        ).count()
```

Comment:
One thought here is that maybe this should happen specifically in the creation function. Since we have the de-duping, it's valid to create a monitor on an existing url even if we're at the limit, since it doesn't cause any more hits to that url.

Maybe the create function could raise an exception that we catch and translate in save?

Maybe also just an edge case and not a big deal

Reply:
oh right, hmm. But you can edit a monitor to change the url as well right. Yeah I guess it could be an exception.

Let's not worry about it for now.